internal/client/gitea.go

```diff
@@ -51,6 +51,32 @@ func NewGitea(ctx *context.Context, token string) (Client, error) {
 	return &giteaClient{client: client}, nil
 }
 
+// CloseMilestone closes a given milestone.
+func (c *giteaClient) CloseMilestone(ctx *context.Context, repo Repo, title string) error {
+	milestone, err := c.getMilestoneByTitle(repo, title)
+
+	if err != nil {
+		return err
+	}
+
+	if milestone == nil {
+		return ErrNoMilestoneFound{Title: title}
+	}
+
+	closedState := string(gitea.StateClosed)
+
+	opts := gitea.EditMilestoneOption{
+		Deadline:    milestone.Deadline,
```

Comment:
Probably not a big deal when closing the milestone, but I think this is prone to a race condition where the milestone's data gets modified out of bound before the modified version is sent back.

I suppose we are limited by the SDK/API here anyway?

Reply:
I was operating under the presumption that leaving pointer fields in the update options empty could blank them out in the remote object -- I do not know for sure if that's the actual case though. Happy to change this any which way.

Reply:
Aye, sorry I meant racing another request/human outside of GoReleaser with GoReleaser, rather than the code itself being racy within GoReleaser. In other words I wondered if there was a way of sending a single "partial update" request as opposed to "get everything" -> "update everything"?

Reply:
Not that I'm aware of 😅

Reply:
👌LGTM then